### ISSUE TITLE
How to add metadata or header ?

### ISSUE THREAD

**Issue by technomaniac** [gogo/letmegrpc#51]

**Comment by awalterschulze** [comment]
https://github.com/gogo/letmegrpc/blob/master/form/form_test.go

**Comment by scbizu** [comment]
Hi @awalterschulze , I read the `CreateCustom` in `form.go`,and still have some questions. When we setup context metadata/header in grpc context , we will not define these kind of fields explicitly.

In my way, I will prefer to add an additional field(say as `context`),this field name and value is also customized, the fake code will look like this: 

```go
	contextString := req.FormValue("context")
	contextStringMap := make(map[string]string)
	if len(contextString) > 0 {
		err := encoding_json.Unmarshal([]byte(contextString), &contextStringMap)
		if err != nil {
			if err != io.EOF {
				w.Write([]byte("<div class=\"alert alert-danger\" role=\"alert\">" + err.Error() + "</div>"))
				return
			}
			w.Write([]byte("<div class=\"alert alert-danger\" role=\"alert\">" + err.Error() + "</div>"))
		}
	}
	w.Write([]byte(FormClient_RPCXXXXXX))
	if !someValue {
		w.Write([]byte(Footer))
		return
	}
	for metaKey, metaValue := range contextStringMap {
              // do some meta match here
        }
```

**Comment by awalterschulze** [comment]
I haven’t used grpc in a few years, so I fear I am missing some context ;)

sorry that is a bad joke, but I am struggling to know which problem you are
trying to solve

**Comment by scbizu** [comment]
Haha , you can check the [grpc-go's oauth example](https://github.com/grpc/grpc-go/blob/master/examples/oauth/server/main.go#L96) for the grpc context metadata.

**Comment by awalterschulze** [comment]
What do you want to do with context in letmegrpc?

**Comment by scbizu** [comment]
we will use this kind of data for auth and trace and other middleware operations

**Comment by awalterschulze** [comment]
So do you want to print it out as a response or read in as input or
programmatically provide it?

**Comment by scbizu** [comment]
Read it as input

**Comment by scbizu** [comment]
yep, so I fork another letmegrpc to our private repo, and will do some special edition XD

**Comment by awalterschulze** [comment]
Hmmm yeah or you show me how you make it work and contend to take over
letmegrpc :) ?

**Comment by scbizu** [comment]
aha  still WIP
